coffea/processor/executor.py

```diff
@@ -843,9 +852,20 @@ def _get_cache(strategy):
     return cache
 
 
+class ParquetFileContext():
+    def __init__(self, filename):
+        self.filename = filename
+
+    def __enter__(self):
```

Comment:
Here the file should be opened. The purpose of the context manager is to ensure we close the file and hopefully deallocate the arrays that were read from it regardless of how the processing went or if a reference to the NanoEvents objects persists.

Reply:
The dataset API requires file paths as input. It does not accept open file handles. So, that's why I created a dummy ContextManager.